### DISCUSSION TITLE
Koriki 1.0 for RG35XX

### DISCUSSION THREAD

**Discussion by acmeplus** [rg35xx-cfw/Koriki#80] [Announcements]
### Installation:
* Download the image file from this page (``koriki-rg35xx-20231011-v1.0.img.gz``)
* On your Linux/Mac/PC use a tool like balena etcher to flash the image to a SDCARD (at least 8GB size)
* Once flashed, insert the card into your RG35XX and power it up (you may require to press reset or long press power button)
* The first boot will take longer while the installation expands the SDCARD and copies required files into the third partition (SHARE)
* Enjoy!

### IMPORTANT:
* If you are reusing an existing card on the second SDcard port that contains with roms and other content from a previous Korki or batocera installation, you should rename or remove the system folder so the new configuration files are properly copied. 

### Change log:

* Synchronized code base to batocera devel (V39)
* Updated all cores to the latest versions
* Updated all standalone emulators (PPSSPP, flycast, etc)
* Added shutdown after removing power cable during charging
* Added initial suspend mode (low enery userspace mode, not kernel based, not hibernation): Short press Power button to enter/exit suspend mode
* Added shutdown shortcut: press volume+ + Power button for shutdown
* Brightness level is now saved
* All hardware revisions should work without requiring bootfixes
* Added new theme (Airon Theme) Thanks @SzalikDesigns!
* Added stylus support for NDS (R2 + DPAD for movement, R2 + A to click)
* Ports:
  * Added VVVVVV port
  * Added back SpaceCadet port
  * @JohnnyOnFlame ports (MalditaCastilla, Spelunky, CurseOfIssyos, ShovelKnight) should now work and exit correctly
  * Updated OpenBOR versions, default is now 6510
  * Added additional ports/emulators: Amiberry, OpenJazz, CDOGS, ECWOLF, eDuke32, Fallout 1 & 2, Fury, OpenTyrian, Simcoupe, etc. Note that all these require data file, refer to batocera pages for those ports for information about required data files
* Other changes:
  * ODCommander keys are now working
  * OpenBOR default keys configured
  * Updated default PPSSPP standalone mappings (thanks @S1eepy)

### Default shortcuts:
* ``MENU`` + ``Volume Up`` -> Brightness Up
* ``MENU`` + ``Volume Down`` -> Brightness Down
* ``POWER Button`` Short press -> Suspend/Resume 
* ``Volume Up`` + ``POWER Button`` -> Shutdown
* Retroarch shortcuts:
  * ``MENU`` + ``A``: Pause
  * ``MENU`` + ``B``: FastForward Toggle
  * ``MENU`` + ``X``: Menu
  * ``MENU`` + ``Y``: Reset Content
  * ``MENU`` + ``L1``: Save Save State
  * ``MENU`` + ``R1``: Load Save State
  * ``MENU`` +`` L2``: Next Save State
  * ``MENU`` + ``R2``: Previous Save State
  * ``START`` + ``SELECT``: Quit Game   
* Many standalone emulators can be exited with ``START`` + ``SELECT``

Special Thanks:
* @JohnnyOnFlame for the SDL2 patches, ports, and general help and insight
* [@Neocvera ](https://www.youtube.com/@neocvera) for testing, suggestions, and videos
* @SzalikDesigns for the new Airon Theme
* [Rparadise-Team,](https://github.com/Rparadise-Team) for the original Koriki for the Miyoo Mini and their help
* @albgr1979 for the ATC Koriki themes
* [@S1eepy](https://www.youtube.com/@S1eepy1) for testing, and emulator configuration files
* @adixal for the joyaudio, and in general for all his great work and ideas on MuOS and helping the RG35XX community 
* Retro Handhelds and Retro Game Handhelds discord communities (e1000, Xquader, Aveferrum, Joe Staff,... and many more)
* Everyone who has supported me in [Ko-Fi](https://ko-fi.com/acmeplus)

<hr /><em>This discussion was created from the release <a href='https://github.com/rg35xx-cfw/Koriki/releases/tag/koriki_rg35xx_v1.0'>Koriki 1.0 for RG35XX</a>.</em>

**Comment by ThOoM013** [comment]
Thank you sir and all others involved for you're hard work. Lets check this. bad boy out!!

**Comment by netsba** [comment]
Thank you very much for the work you have done. It's going to make my life easier

**Comment by korikirg35xx** [comment]
Hi.I put koriki 1 (rg35xx) today. but i can't shutdown the device . I charge the battery ,choose shutdown from menu but now i turn on the device and the battery is empty

**Reply by acmeplus** [reply]
> Hi.I put koriki 1 (rg35xx) today. but i can't shutdown the device . I charge the battery ,choose shutdown from menu but now i turn on the device and the battery is empty

Thanks for the report. How did you check the battery is empty? How are you restarting the device? Was the battery fully charged before shutdown?

**Reply by korikirg35xx** [reply]
Thanks for your response.I check the battery from the battery icon shows in menu.The battery was fully charged. Every time i turn on the device with long press power button and turn off with shutdown from the menu. Also there is no led indicator when charge (this happens in ver 1.0) .
P.S : I fully charge the battery at the night . Now i turn off the device from menu (shutdown) . I will turn on in a few hours to check

**Reply by demidy** [reply]
I got the same issue

**Reply by demidy** [reply]
And mine is not charging anymore

**Reply by paladinperry** [reply]
same issue, koriki wont turn off - long press no - short press no - has to drain the battery
having issues charging it now also - have to remove SD card and try cyling with power cable over and over

Thank you for the hard work. I exclusively use your koriki and will continue to do so, thats why this is such a big issue for me

**Comment by MancV21** [comment]
Hi, first of all, THANK YOU for this.

Someone mentioned on Reddit Koriki now has native PICO support, is this right? If so, how do I go about it? Just copy my PICO files from their website into the folder? Does it support Splore?

**Reply by acmeplus** [reply]
You can either use fake08 core in retroarch, or you can use your pico8 binary purchased from lexaoffle. However, keep in mind that without internet access Splore functionality will be limited.

To get pico8 to work, add a new launcher script to [SHARE]/roms/ports/Pico8.sh with the following content:
```bash
#!/bin/sh

export SDL_GAMECONTROLLERCONFIG=$(grep "RG35XX" "${HOME}/.config/gamecontrollerdb.txt")

cd /userdata/roms/pico8
/userdata/roms/pico8/pico8_dyn -joystick 0 -splore
```

Then make sure you copy the raspberry version of pico8 to the /userdata/roms/pico8/ folder (you should have pico8_dyn there)

**Comment by dmauri82** [comment]
Hi all! I have a RG35 with a different motherboard and now the boot image and buttons mapping works properly!
I detected a lower performane in GBA emulator (playing fo example kukurin which is not a specially demanding game)
Kokori uses all the CPU resources or I can increase the cpu clock? (like garlic os). I don't see any option neither menu...
Tks guys for all your effort!

**Reply by acmeplus** [reply]
Will check if there has been an issue with mgba core and update/revert its version if needed.

**Reply by chuck-beowulf** [reply]
There's definitely an issue with mGBA; I also had problems with it running slowly with messed-up music and no fast-forwarding. When I manually switched the core to gpSP it worked.

**Reply by acmeplus** [reply]
Back in 0.9.1/0.9.2 mGBA was held back a few versions due to performance issues. Will check if it can be reverted to that version.

**Comment by rodrigovrlucena** [comment]
@acmeplus Thank you very much for this 🎉 I can translate Koriki into portuguese - Brazil, i just needed to know where the menu files are. Can you help me?

**Reply by acmeplus** [reply]
Thanks, but unfortunately simplemenu does not support translation. You can translate the themes, but not the menu text without recompiling the application. 

Simplermenu+ is in development and will have international strings support from the start.

**Comment by acmeplus** [comment]
For those of you testing Drastic, you need an update that will be in the next version. First update the launcher in ``[SHARE]/system/.simplemenu/drastic_sa`` to match this file:
[drastic_sa](https://raw.githubusercontent.com/rg35xx-cfw/buildroot/d830b4d3887d4e113cbed77f16faea1b204564e7/board/rg35xx/fsoverlay/usr/share/simplemenu/.simplemenu/launchers/drastic_sa)
And replace ``[SHARE]/system/configs/drastic/config/drastic.cfg`` with this file:
[rastic.cfg](https://raw.githubusercontent.com/rg35xx-cfw/buildroot/5ce221d5972b10701696d1645b1567d5372da651/board/rg35xx/fsoverlay/usr/share/drastic/config/drastic.cfg)

**Reply by joshg253** [reply]
Assume you mean? [SHARE]/system/.simplemenu/**launchers**/daphne_sa

**Reply by acmeplus** [reply]
> Assume you mean? [SHARE]/system/.simplemenu/**launchers**/daphne_sa

Yes. The launchers path was missing. But no, it's not Daphne.

**Comment by smeejaytee** [comment]
looks promising thank you and everyone involved for your time and effort, is it possible to update from 9.2 or would I need to start from scratch

**Comment by HDPyamato** [comment]
Hello! Thanks for the very nice work, can't wait to see next. 
Also, been testing since 0.7 and never get the img/thumbnails of games working... I try many folder managment but always no picture, even with Old Towers included. Is it just me or.. ?

**Reply by SzalikDesigns** [reply]
Try put them in folder **Imgs** and name them same as rom with png format, and they will work fine

**Comment by smeejaytee** [comment]
cant see master system games in consoles is this a bug or something I've one I'm not a noob and it worked fine on 9.2 Tia

**Reply by chuck-beowulf** [reply]
I had to rename my "ms" roms folder to "mastersystem" and then it recognized them.

**Comment by chuck-beowulf** [comment]
I've been using GarlicOS and just set up a new card with this, and I LOVE the setup. Pretty, organized, generally works well. I'm astounded at how well N64 is running on this and that PSP runs at all. The issues I found:
- mGBA problem noted above.
- Pokemini and Virtual Boy games won't run--they try to open and then bounce right back to the menu. DS games opened a black screen and froze, forcing a hard reboot. Pico-8 games didn't map the controller, but I see there's a fix for that above.
- PSP doesn't have any hotkeys or menu keys mapped. I had to hard reboot to get out of a game.
- SNES crashes out of games if you try to enable a cheat code. (This was an issue in GarlicOS, too, and I have no idea why because cheat codes worked fine everywhere else I tried them.)
Seriously, though: This is great work. I'll probably switch to it full-time even if this above issues aren't fixed (except maybe for GBA games).

**Reply by acmeplus** [reply]
Thanks for the report, a few answers:
* For GBA, as explained above, I've to check if the older version performs better. I had Mario Kart running without issues in 0.9.2.
* For DS games check [this comment](https://github.com/rg35xx-cfw/Koriki/discussions/80#discussioncomment-7558339).
* For PSP you can start it manually from the Apps folder and then configure the controls and save them
* About poke mini and VB, just checked and they were not included in 1.0, will add them back in the next update.

**Reply by benni092** [reply]
The Problem with MGBA can be vsync. I make a short demo video to show you what i mean. You can hear it on audio when i turn on vsync it begins to crackle and then i turn off vsync and the audio becomes normal again.

https://github.com/rg35xx-cfw/Koriki/assets/122157977/3cc15cf6-81a2-4809-93dc-ea49c3c2a89f

**Reply by chuck-beowulf** [reply]
Having fiddled with things this morning: Turning off vsync does seem to solve the GBA problem (fast-forward still isn't particularly fast, but that's a more minor issue). Mapping the L2 key to pause addressed my PPSSPP issue. I tried doing the Drastic fix for DS, but now instead of freezing, games just dump me back to the menu. So now I'll eagerly await the next version for that fix, Pokemini, Virtual Boy, and Pico-8 controls. (And in the meantime, this is great and I'm...going to recharge my RG35XX so I can play more!)

**Reply by acmeplus** [reply]
There's an update coming up with a much faster gba that seems to perform well. That should address that issue.

For the virtualboy, there's an error in the consoles.in (``[SHARE]/system/.simplemenu/section_groups/consoles.ini``). Edit that file and update the VIRTUAL BOY section to match this:
```ini
[VIRTUAL BOY]                                                                                                       
execs = /userdata/system/.simplemenu/launchers/vb_libretro                                                          
romDirs = /userdata/roms/virtualboy/,/userdata/Roms/VB/                                                             
romExts = .vb,.vboy,.bin,.zip,.7z 
```

**Reply by chuck-beowulf** [reply]
Yes! Virtual Boy works now. It's running a little sluggish and fast-forward doesn't work, but it's playable. (And much better than the half-speed it runs at in GarlicOS) I suspect this is another case where it just needs a settings tweak.

Other things: I noticed that SegaCD was among the available consoles, but it won't run games. I also got around to trying arcade emulators, and while MAME works great, FBA won't run games, and Neo-Geo tries to open them but then gives a white screen with a tiny error message I can't read. (Maybe the problem is my rom selection? I don't know.) 

I'd also like to applaud you for being more responsive than 90% of the developers I have to deal with on a professional basis, especially for something like this when I (and likely the rest of your audience) will happily wait as you develop on your own time.

**Reply by acmeplus** [reply]
Will check SegaCD. For VirtualBoy make sure you have video thread set to on.

**Reply by chuck-beowulf** [reply]
Okay, so, I installed v1.01, and I turned on video threading for Virtual Boy (which makes it work slightly better, but still not perfect and fast-forward doesn't do anything). mGBA definitely runs much better now. 

Remaining issues: Pokemini still doesn’t work. SegaCD still doesn’t work. I swapped my FBA folder to FBN, and Final Burn Neo runs for at least some of the roms I have. Neo-Geo still gives me the white error screen. I had notes that Master System cheats worked before; now they dump me back to the menu (and SNES cheats still don't work)--I had to recopy the Liberto database after I deleted the system folder for the new version, so I'm wondering if something went wrong there. Also, I've been trying to map the Pico-8 controls, but can't figure out all the steps.

**Reply by chuck-beowulf** [reply]
Oh, and DS works for me now! I was playing some Kirby last night, and the sound was a little crackly but it played fine, which is really amazing on a device like this.

**Comment by benni092** [comment]
Jeah i think i have found the problem with fast forward. In Retroarch go to settings Frame Throttle and change Fast-Forward Rate to 50. It was on 0.  For gba fast foward use gpsp, with mgba there is still no change in speed.
![IMG_20231116_232005](https://github.com/rg35xx-cfw/Koriki/assets/122157977/ff20e62b-9aee-4bb5-92d9-fabd7e0c36e6)

**Comment by acmeplus** [comment]
Koriki has been updated to 1.0.1: https://github.com/rg35xx-cfw/Koriki/releases/tag/koriki_rg35xx_20231116

Major changes are new mgba core and headphone fixes. Additional notes in the release.

**Comment by demidy** [comment]
i install the 1.0.1 version and the LED lights are not on, and now i don't know when the device is fully charged, can you help me pls? and i notice that the psp performance is slightly worse than in the 0.9.2 version

**Reply by korikirg35xx** [reply]
I have the same.There is no led light in charging.

**Reply by demidy** [reply]
i just plug it and leave for like 2 or 3 hours and then i play, seems charged but drains really fast

**Comment by maximtyw** [comment]
Thanks alot for the great release. I found it amzing to be able to play NDS on this little device. i have a few questions through:

1. NDS in v1.01 now the key mapping is working we can use stylus with the combo key. but I would like to has additional combo key (maybe like menu+L2 or R2 etc) to swap the screen ( 2screen vertical/horizontal). I understand that we not able to assign the combo-key in the normal drastic menu. but i think we can dirrectly assign the key (the correct number?) in the drastic config file. is there any documenentation or method to find out what number (combo-key) in koriki or drastic? 

2. I saw that Drastic on other CFW is able to has a dual sreen with bigger top screen. Cant this to be done on Koriki? is there any file in dractic folder we can play around to get the bigger top screen set-up?


Again thanks alot for the great CFW.

**Comment by maximtyw** [comment]
Thanks for the advice. i will definitely check it out.

**Comment by job1022** [comment]
Will atomiswave be included? I think there's some systems that although have a folder for roms, don't show in menus.

**Comment by M4d3v1l5** [comment]
hi, i am curently using version 1.0.1, using updated file instaled on version 1.0. I found some bug while using it :

1. Battery level is somehow stuck in 32~%, even after charged over 2 hours. I can't check if its actually use the full battery capacity or not because now the red led indicator didnt showup anymore. On ver 9.2 the red light & battery indicator is still work properly, starting from build 1.0 its now gone.
2. The sound is now in the increment of five (0,5,10,15,20,) so the volume is now either to low or to loud, i suspect that something in the setting folder is messup with the updated file.
3. Playing Gba Movie with any gba core is now become terible experience, for upon booting the Gba Video file now the audio is stuck in max volume, blasting the speaker to the max, even more horible if you use headset/earphone like me. The previous version (9.0~) didnt have this problem, all .gba file play with same audio volume, regardless its content p. Something is definitely goes wrong with some audio setting in this ver. 1.0.1. strangely its only afect Gba Video Rom, most if not all Gba game work fine.

I will try to flash ver 1.0.1 fully on the new sd card (when its arive, shiping got stuck near christmast) to confirm if its only ver 1.0+1.0.1 that messup the setting or also the full 1.0.1 version.

I hope some of this problem above can be checked and hopefully can be fixed on the future version, thanks

**Reply by M4d3v1l5** [reply]
Update :

Gba Video still start with full volume, but pressing any button will bring it back to normal volume level, as workaround i just press A/B as soon as the .gba file load. Maybe the bug is in the retroarch itself and not from the koriki Os? as some gba game (Tested: B-daman game) also sometime start at full volume while sometimes it didnt have any problem whatsoever. But the problem presist across in any gba core used, so eh, i dont know?

**Reply by M4d3v1l5** [reply]
Eh, apparently it from the 1.0.1 Os itself as issue #87 state
